shared/src/main/kotlin/com/egm/stellio/shared/model/ExpandedEntity.kt

```diff
@@ -25,4 +28,50 @@ class ExpandedEntity private constructor(
 
     fun compact(): Map<String, Any> =
         JsonLdProcessor.compact(attributes, mapOf("@context" to contexts), JsonLdOptions())
-}
+
+    private fun getAttributeOfType(type: AttributeType) =
+        attributes.filterKeys {
+            !listOf(NgsiLdParsingUtils.NGSILD_ENTITY_ID, NgsiLdParsingUtils.NGSILD_ENTITY_TYPE).contains(it)
+        }.mapValues {
+            NgsiLdParsingUtils.expandValueAsMap(it.value)
+        }.filter {
+            NgsiLdParsingUtils.isAttributeOfType(it.value, type)
+        }
+
+    /**
+     * Gets relationships of [entity]
+     * Relationships can either be direct, or from a property.
+     *
+     * @return a subset of [entities]
+     */
+    fun getRelationships(): List<String> =
```

Comment:
So, well, I think it deserves a better name. Maybe something like `get(All)LinkedEntitiesIds`?

Reply:
I agree that the name isn't really clear. I'll take getLinkedEntitiesIds :)